File: python/a03.py
# A03 Two Cards
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38326075

n, k = map(int, input().split())
p = list(map(int, input().split()))
q = list(map(int, input().split()))

for pp in p:
    for qq in q:
        if pp + qq == k:
            print('Yes')
            break
    else:
        continue
    break
else:
    print('No')

File: rust/tessoku-book/src/bin/a03.rs
// A03 Two Cards
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38326104

use proconio::input;

fn main() {
    input!{
        n: u32,
        k: u32,
        p:[u32; n],
        q:[u32; n],
    }
    let mut found = false;
    for pp in &p {
        for qq in &q {
            if pp + qq == k {
                found = true;
            }
        }
    }
    if found {
        println!("Yes");
    } else {
        println!("No");
    }
}
